Sources/TensorFlow/Optimizer.swift

```diff
@@ -294,3 +294,62 @@ public class RiemannSGD<Model: Layer, Scalar: FloatingPoint>: Optimizer
         model = model.moved(along: learningRate * (.zero - direction))
     }
 }
+
+/// AdaGrad optimizer.
+/// Individually adapts the learning rates of  all model parameters by scaling them inversely proportional to 
+/// the square root of the sum of all the historical saquared values of the gradient.
+/// 
+/// Reference: ["Adaptive Subgradient Methods for Online Learning and Stochastic Optimization"](
+///  http://www.jmlr.org/papers/volume12/duchi11a/duchi11a.pdf)
+/// 
+public class AdaGrad<Model: Layer>: Optimizer
+    where Model.AllDifferentiableVariables == Model.TangentVector {
+    /// The learning rate.
+    public var learningRate: Float
+    // Rho is the smoothing factor. Typical values are 0.5, 0.9, 0.99. (smooth over 2, 10, 100 examples)
```

Comment:
* Use 3 slashes for doc comments.
* Add backquotes around numeric values in comments. 

```suggestion
    /// The smoothing factor (ρ). Typical values are `0.5`, `0.9`, and `0.99`, for smoothing over 2,
    /// 10, and 100 examples, respectively).
```